src/core/api/video_track_manager.ts

```diff
@@ -0,0 +1,350 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This file is used to abstract the notion of video track-switching
+ * for an easier API management.
+ */
+
+import arrayFind from "array-find";
+import { Subject } from "rxjs";
+import log from "../../log";
+import { Adaptation, Period } from "../../manifest";
+import arrayIncludes from "../../utils/array-includes";
+import SortedList from "../../utils/sorted_list";
+
+// video track returned by the VideoTrackManager
+export interface ILMVideoTrack {
+  id : number|string;
+}
+
+// video track from a list of video tracks returned by the VideoTrackManager
+export interface ILMVideoTrackListItem extends ILMVideoTrack {
+  active : boolean;
+}
+
+// stored video informations for a single period
+interface ILMPeriodVideoInfos {
+  adaptations : Adaptation[];
+  adaptation$ : Subject<Adaptation|null> ;
+}
+
+// stored informations for a single period
+interface ILMPeriodInfos {
+  period : Period;
+  video? : ILMPeriodVideoInfos;
+}
+
+/**
+ * Manage video tracks for all active periods.
+ *
+ * Most methods here allow to interact with the first chronologically added
+ * Period.
+ *
+ * Roles for subsequent periods are also chosen accordingly.
+ * @class VideoTrackManager
+ */
+export default class VideoTrackManager {
```

Comment:
Why not including this code in the ``LanguageManager`` - and renaming it something like ``TrackManager``?

It seems that both the ``VideoTrackManager`` and ``LanguageManager`` do the exact same thing